From 05009b32ef027f06f604f491e4e95e283d0dfc73 Mon Sep 17 00:00:00 2001
From: Alex Schultz <aschultz@redhat.com>
Date: Wed, 21 Mar 2018 15:07:43 -0600
Subject: [PATCH] Add openvswitch element back in

The change Ie4f5d771a16ea453b470be8ea103b2bde4aa892a switched
os-net-config to be installed as a package rather than an element.
Unfortunately the openvswitch service was being configured via the
openvswitch dependency from the os-net-config element. The switch to use
the package lost the automatic starting of openvswitch on the nodes
which has manifested itself as intermittent issues on compute nodes when
network isolation is not deployed.

Change-Id: I0f865d4811919c19577c75615b66d7d8a1e685d3
Partial-Bug: #1757111
---
 image-yaml/overcloud-hardened-images.yaml  | 1 +
 image-yaml/overcloud-images.yaml           | 1 +
 image-yaml/overcloud-odl-rhel7.yaml        | 1 +
 image-yaml/overcloud-realtime-compute.yaml | 3 ++-
 4 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/image-yaml/overcloud-hardened-images.yaml b/image-yaml/overcloud-hardened-images.yaml
index 43e97f2..bf6316f 100644
--- a/image-yaml/overcloud-hardened-images.yaml
+++ b/image-yaml/overcloud-hardened-images.yaml
@@ -4,6 +4,7 @@ disk_images:
     type: qcow2
     elements:
       - dhcp-all-interfaces
+      - openvswitch
       - overcloud-agent
       - overcloud-full
       - overcloud-controller
diff --git a/image-yaml/overcloud-images.yaml b/image-yaml/overcloud-images.yaml
index 81c71c2..12202e5 100644
--- a/image-yaml/overcloud-images.yaml
+++ b/image-yaml/overcloud-images.yaml
@@ -5,6 +5,7 @@ disk_images:
     elements:
       - baremetal
       - dhcp-all-interfaces
+      - openvswitch
       - overcloud-agent
       - overcloud-full
       - overcloud-controller
diff --git a/image-yaml/overcloud-odl-rhel7.yaml b/image-yaml/overcloud-odl-rhel7.yaml
index 5792eb3..efaaad1 100644
--- a/image-yaml/overcloud-odl-rhel7.yaml
+++ b/image-yaml/overcloud-odl-rhel7.yaml
@@ -8,6 +8,7 @@ disk_images:
            - sysctl
            - baremetal
            - dhcp-all-interfaces
+           - openvswitch
            - os-collect-config
            - puppet-modules
            - stable-interface-names
diff --git a/image-yaml/overcloud-realtime-compute.yaml b/image-yaml/overcloud-realtime-compute.yaml
index 4e39d40..4b6a3a5 100644
--- a/image-yaml/overcloud-realtime-compute.yaml
+++ b/image-yaml/overcloud-realtime-compute.yaml
@@ -5,10 +5,10 @@ disk_images:
     elements:
       - baremetal
       - dhcp-all-interfaces
+      - openvswitch
       - overcloud-agent
       - overcloud-compute
       - puppet-modules
-      - os-net-config
       - stable-interface-names
       - grub2
       - element-manifest
@@ -28,6 +28,7 @@ disk_images:
       - ntp
       - device-mapper-multipath
       - openstack-heat-agents
+      - os-net-config
       - jq
     options:
       - "--min-tmpfs=7"
-- 
2.13.6

